### PR TITLE
ENH: Add Better Support For Tiled TIFFs

### PR DESCRIPTION
**Pull request by John-P**
- Add a check for tiles in a TIFF file when opening 
- Use OpenSlide to read a TIFF if it has tiles instead of OpenCV (VirtualWSIReader).
- Add a fallback to tifffile if it is tiled but openslide cannot read it (e.g. jp2k or jpegxl tiles).

### CODE REVIEW

**Comment by John-P** [comment]
This is currently failing due to now supporting more TIFF formats, and we don't currently have an example for which this would fail. Therefore, either the test coverage will be reduced which will fail the PR or the test will fail which will fail the PR. I will look into generating a test image where this should fail.

**Comment by John-P** [comment]
I have made a simple test image which is a copy of CMU-1-Small-Region.svs but with two tiled pages and no resolution metadata. This is used as an example fail case, as we currently don't support multipage generic tiled TIFF. Only known TIFF formats such as SVS, NDPI, OME-TIFF, and generic single-page tiled TIFFs. This could be extended to single series, e.g. one series of pyramid resolutions in future. This test image would also fail that, as there is no subifd tag for reduced resolution set.

File: tiatoolbox/utils/misc.py [hunk 1]
```diff
@@ -756,3 +756,30 @@ def string_to_tuple(in_str):
...
+    """
+    microns_per_unit = {
+        "centimeter": 1e4,  # 10,000
+        "cm": 1e4,  # 10,000
+        "inch": 25400,
+        2: 25400,  # inches in TIFF tags
+        3: 1e4,  # cm in TIFF tags
+    }
+    if units not in microns_per_unit:
+        raise ValueError(f"Invalid units: {units}")
```
Comment (shaneahmed) [inline tiatoolbox/utils/misc.py:784]: A small test can be added with invalid units to raise the error.

**Review by shaneahmed — COMMENTED** [review]
Thanks @John-P 
Please can you add the test as mentioned?

**Comment by shaneahmed** [comment]
@R-J96 As this is related to #314. Please can you review this PR?

**Review by R-J96 — APPROVED** [review]

**Review by shaneahmed — APPROVED** [review]

**Comment by John-P** [comment]
@shaneahmed My current solution to the deepsource path issue is a combination of `# skipcq: PTC-W6004` it ignore generated paths in the whole file:
https://github.com/TissueImageAnalytics/tiatoolbox/blob/44c59b443bf4381892f7bb0487d48bbebc580707/tests/test_annotation_stores.py#L1

and allowing commented out code in tests (vecause flake thinks that skipcq is code!):
https://github.com/TissueImageAnalytics/tiatoolbox/blob/44c59b443bf4381892f7bb0487d48bbebc580707/setup.cfg#L25-L26

**Pull request merged**